### ISSUE TITLE
Stuck on startup

### ISSUE THREAD

**Issue by echosssss** [roflmuffin/CounterStrikeSharp#443]
Hello everyone, I wanted to open a CS2 server and develop plugins, for this reason, I decided to use CSS but I faced an error I couldn't understand.

I installed Metamod and CSS with Runtime according to your docs, and then CSS successfully started without any plugin but when I added HelloWorldPlugin to the `plugins` folder, I was stuck on startup :/ You can see below where I am stuck, I waited for hours and it was still the same

![image](https://github.com/roflmuffin/CounterStrikeSharp/assets/168978699/716733fd-6d09-49b6-ab9d-aeb5a337acf2)

## HelloWorldPlugin.cs
```cs
using CounterStrikeSharp.API.Core;

namespace HelloWorldPlugin;
public class HelloWorldPlugin : BasePlugin
{
    public override string ModuleName => "Hello World Plugin";

    public override string ModuleVersion => "0.0.1";

    public override void Load(bool hotReload)
    {
        Console.WriteLine("Hello World!");
    }
}
```
![image](https://github.com/roflmuffin/CounterStrikeSharp/assets/168978699/caf9d5e6-572b-4c67-a771-0846ca9ebe4e)

## My File Tree on The Server
![image](https://github.com/roflmuffin/CounterStrikeSharp/assets/168978699/5897bffc-28df-4b51-9821-9ad46cc9366b)

## My Server
Ubuntu 22.04.4 on Oracle Cloud Ampere VM.Standard.A1.Flex

![image](https://github.com/roflmuffin/CounterStrikeSharp/assets/168978699/98d428bf-85fa-4f7e-ba17-b34074e9d0e7)

### start.sh

```sh
/home/ubuntu/Steam/steamapps/common/cs2/game/bin/linuxsteamrt64/cs2 -dedicated -secure -usercon -maxplayers 10 -port 31313 +hostname "Game Server" +game_alias competitive +map de_inferno +sv_setsteamaccount token +exec steamapps/common/cs2/game/csgo/cfg/server.cfg
```

**I don't know what I did wrong and what I missed, thanks for your help ❤️**

**Comment by echosssss** [comment]
Yes, it passes where I am stuck without any plugin and successfully starts the server:

![image](https://github.com/roflmuffin/CounterStrikeSharp/assets/168978699/ef8b9d2f-bb2b-44bb-8c21-2a1c7cc52021)

And I can see it with `meta list` command:

![image](https://github.com/roflmuffin/CounterStrikeSharp/assets/168978699/d10bcda8-2ca5-4f0b-a61c-4d5c5bec9096)

Other than this, I tried two different retake plugins written for CS# before opening this issue. The result is the same, I am stuck after "Admin command overrides file not found. Skipping admin command overrides load." log. Therefore I thought it might be about CS#. I faced issues like this because of mostly ARM arch but in this situation, there may be many reasons because I don't take any errors.

Thanks for your help.

**Comment by elgato69** [comment]
This is related to #180 where running CSS under any sort of emulation layer on AArch64 hosts results in this hanging behavior.

From my observations the following causes the issue
- Issuing any CSS commands at the server console 
- Attempting to load CSS with any plugin installed (eg. @echosssss "Hello World" plugin)

Doing any of these causes a server hang leading to the watchdog stopping the server

I have further tested under the following environments

- Pure x86 chroot via qemu-binfmt method
- FEX-Emu, both through FEXInterpreter and FEXBash

My system is also Ubuntu 22.04 on Oracle Ampere A1. It is a very frustrating issue since the server works perfectly fine under the emulation layer with only Metamod, and while CSS loads and appears in the output of `meta list` trying to interact with it in any way just hard crashes. I have been testing with every new CSS version and FEX update with no joy so far. I would really like to see this issue fixed as it has been an ongoing problem stopping me from hosting my CS2 server.

**Comment by elgato69** [comment]
Hi, apologies to respond on such an old issue.

Just wanted to note that I found a workaround for this. I can't remember exactly where I found this but was browsing around recently while trying to get this to work again.

It turns out if you set the environment variable `DOTNET_EnableWriteXorExecute` to 0, the server will function correctly

I simply added `export DOTNET_EnableWriteXorExecute=0` to my server startup script and I was able to launch the server with CounterStrikeSharp and load plugins without any issues so far.

Hope this helps someone
